app/src/main/java/com/asfoundation/wallet/topup/TopUpFragmentPresenter.kt

```diff
@@ -158,4 +186,116 @@ class TopUpFragmentPresenter(private val view: TopUpFragmentView,
           }
         }
   }
+
+  private fun handleInvalidValueInput(packageName: String, topUpData: TopUpData) {
+    if (topUpData.currency.fiatValue != DEFAULT_VALUE) {
+      val value =
+          FiatValue(BigDecimal(topUpData.currency.fiatValue), topUpData.currency.fiatCurrencyCode,
```

Comment:
I would be careful with this as this was originally a BigDecimal turned to String and then it’s being converted to BigDecimal again. This already gave us problems once, due to BigDecimal not handling “,” very well